### PR TITLE
Implement Result MarkPreReconcilerFailed to fix upgrade

### PR DESCRIPTION
**Pull request by khrm**
# Changes

This fixes panic during the upgrade from previous and panic in other scenarios during the failure.

error:
```
{"level":"error","logger":"tekton-operator-lifecycle","caller":"tektonresult/tektonresult.go:152","msg":"PreReconciliation failed: TektonResult/pre: installer set not ready, will retry: Install failed with message: reconcile again and proceed","commit":"1d95ed3","knative.dev/pod":"openshift-pipelines-operator-67cdfc78db-p68zm","knative.dev/controller":"github.com.tektoncd.operator.pkg.reconciler.kubernetes.tektonresult.Reconciler","knative.dev/kind":"operator.tekton.dev.TektonResult","knative.dev/traceid":"15984a50-220b-4edb-a5e6-a03295d90cfa","knative.dev/key":"result","stacktrace":"github.com/tektoncd/operator/pkg/reconciler/kubernetes/tektonresult.(*Reconciler).ReconcileKind\n\t/go/src/github.com/tektoncd/operator/pkg/reconciler/kubernetes/tektonresult/tektonresult.go:152\ngithub.com/tektoncd/operator/pkg/client/injection/reconciler/operator/v1alpha1/tektonresult.(*reconcilerImpl).Reconcile\n\t/go/src/github.com/tektoncd/operator/pkg/client/injection/reconciler/operator/v1alpha1/tektonresult/reconciler....
panic: implement me
goroutine 2905 [running]:
github.com/tektoncd/operator/pkg/apis/operator/v1alpha1.(*TektonResultStatus).MarkPreReconcilerFailed(...)
/go/src/github.com/tektoncd/operator/pkg/apis/operator/v1alpha1/tektonresult_types.go:104
github.com/tektoncd/operator/pkg/reconciler/kubernetes/tektonresult.(*Reconciler).ReconcileKind(0xc00075a0b0, {0x3583de0, 0xc0025ee4e0}, 0xc001a6e000)
/go/src/github.com/tektoncd/operator/pkg/reconciler/kubernetes/tektonresult/tektonresult.go:156 +0x2308
github.com/tektoncd/operator/pkg/client/injection/reconciler/operator/v1alpha1/tektonresult.(*reconcilerImpl).Reconcile(0xc0006f4280, {0x3583de0, 0xc0025ee4b0}, {0xc002122556, 0x6}) 
```

This issue was introduced in #1961

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Run `make test lint` before submitting a PR
- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Review by khrm — COMMENTED** [review]
/kind bug

**Review by vdemeester — APPROVED** [review]

File: pkg/apis/operator/v1alpha1/tektonresult_types.go [hunk 1]
```diff
@@ -99,9 +99,12 @@ type TektonResultStatus struct {
...
 
-func (in *TektonResultStatus) MarkPreReconcilerFailed(s string) {
-	//TODO implement me
-	panic("implement me")
+func (trs *TektonResultStatus) MarkPreReconcilerFailed(msg string) {
+	trs.MarkNotReady("PostReconciliation failed")
+	resultsCondSet.Manage(trs).MarkFalse(
+		PreReconciler,
+		"Error",
+		msg)
```
Comment (jkandasa) [inline pkg/apis/operator/v1alpha1/tektonresult_types.go:107]: @khrm edit the code after paste :wink:

Comment (khrm) [reply pkg/apis/operator/v1alpha1/tektonresult_types.go:107]: Haha. Done. :)

**Pull request merged**